### ISSUE TITLE
clangd give bogus errors

### ISSUE THREAD

**Issue by solotim** [dense-analysis/ale#2938]
Vi IMproved 8.0 (2016 Sep 12, compiled May 13 2019 20:18:02)
ALE: 2019/12/16 pull
OS: Linux xxx 3.10.0-957.5.1.el7.x86_64 #1 SMP Fri Feb 1 14:54:57 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux  (CentOS)
Clang: clang version 8.0.0 (tags/RELEASE_800/final)

I've been using ALE for a few years, mostly use clang-check for syntax check. Works great. Recently want to try LSP, so changed the linter to clangd. Auto-completion, syntax checking, all are working, except one thing: clangd reports bogus errors; for example it doesn't seem to understand "friend", or doesn't recognize overloaded methods etc. All of them are bogus since the file compiles just fine and clang-check doesn't complain about anything. Both clang-check and clangd are from same build. 

I can only think of one thing: I have to pass extra arguments to clang-check to make it work, while I don't know if I should and/or how to do the same thing on clangd:

```
     let g:ale_cpp_clangcheck_options =
      \ ' -extra-arg -stdlib=libstdc++' .
      \ ' -extra-arg -cxx-isystem' .
      \ ' -extra-arg /<path>/gcc-6.2.0/include/c++/6.2.0/' .
      \ ' -extra-arg -cxx-isystem' .
      \ ' -extra-arg /<path>/gcc-6.2.0/include/c++/6.2.0/x86_64-pc-linux-gnu/'
```

I tried set something in g:ale_cpp_clangd_options, but I found whatever I set there it makes clangd failed to start. Can't tell much from ALEInfo, as the command history is empty. 

Please suggest. Thank you!

**Comment by solotim** [comment]
I have figured out the reason. I need to post-process my compile_database.json to add those extra args to it, then my clangd can understand the cxx-isystem from specific gcc version. 

But the second problem of it, i.e. setting anything to g:ale_cpp_clangd_options invalidates clangd startup is a mystery, which I think is a bug in ALE. Could anyone help reproduce it?

**Comment by solotim** [comment]
Ok, the reason of the 2nd problem is simply I forgot put compile_database.json to parent directory of the source. 

Then why `clangd -pch-storage=disk -compile-commands-dir=build` doesn't help clangd find the json?

**Comment by w0rp** [comment]
There's an existing issue somewhere for not reading `compile_commands.json` from `build`, etc.
